### PR TITLE
[feat]curvefs/client:warmup metric

### PR DESCRIPTION
**Pull request by Cyber-SiKu**
### What problem does this PR solve?

Issue Number: #xxx 

Problem Summary:

### What is changed and how it works?

What's Changed:

How it Works:

Side effects(Breaking backward compatibility? Performance regression?):

### Check List

- [ ] Relevant documentation/comments is changed or added
- [ ] I acknowledge that all my contributions will be made under the project's license

### CODE REVIEW

File: curvefs/src/client/warmup/warmup_manager.h [hunk 1]
```diff
@@ -214,6 +215,9 @@ class WarmupManager {
         return ret;
     }
 
+    virtual void CollectMetrics(InterfaceMetric *interface, int count,
```
Comment (wu-hanqing) [inline curvefs/src/client/warmup/warmup_manager.h:218]: why make this a virtual function?

File: curvefs/src/client/metric/client_metric.h [hunk 2]
```diff
@@ -294,6 +294,16 @@ struct S3ChunkInfoMetric {
     S3ChunkInfoMetric() : s3ChunkInfoSize(prefix, "size") {}
 };
 
+struct WarmupManagerS3Metric {
+    const std::string prefix = "curvefs_warmup";
```
Comment (wu-hanqing) [inline curvefs/src/client/metric/client_metric.h:298]:
```suggestion
    static constexpr const char* prefix = "curvefs_warmup";
```

File: curvefs/src/client/warmup/warmup_manager.h (hunk 1 above)
Comment (Cyber-SiKu) [reply curvefs/src/client/warmup/warmup_manager.h:218]: > why make this a virtual function?

Currently it is the warmup of s3. Considering that the later bs may also support warmup, it is considered to be written like this.

File: curvefs/src/client/metric/client_metric.h (hunk 2 above)
Comment (Cyber-SiKu) [reply curvefs/src/client/metric/client_metric.h:298]: fix

**Comment by Cyber-SiKu** [comment]
cicheck

**Review by wu-hanqing — APPROVED** [review]

**Comment by Cyber-SiKu** [comment]
cicheck

**Comment by Cyber-SiKu** [comment]
cicheck

File: curvefs/src/client/warmup/warmup_manager.h [hunk 3]
```diff
@@ -214,6 +215,9 @@ class WarmupManager {
         return ret;
     }
 
+    void CollectMetrics(InterfaceMetric *interface, int count,
+                                uint64_t start);
```
Comment (SeanHai) [inline curvefs/src/client/warmup/warmup_manager.h:219]: code format

File: curvefs/src/client/metric/client_metric.h [hunk 4]
```diff
@@ -294,6 +294,16 @@ struct S3ChunkInfoMetric {
     S3ChunkInfoMetric() : s3ChunkInfoSize(prefix, "size") {}
 };
 
+struct WarmupManagerS3Metric {
+    static constexpr const char* prefix = "curvefs_warmup";
```
Comment (SeanHai) [inline curvefs/src/client/metric/client_metric.h:298]: Do you test the metric? Will the prefix connect to name with '_', If not, the prefix may 'curvefs_warmup_'

File: curvefs/src/client/warmup/warmup_manager.h (hunk 3 above)
Comment (Cyber-SiKu) [reply curvefs/src/client/warmup/warmup_manager.h:219]: > code format

done

**Comment by Cyber-SiKu** [comment]
cicheck

**Review by SeanHai — APPROVED** [review]

**Comment by Cyber-SiKu** [comment]
cicheck

**Comment by Cyber-SiKu** [comment]
cicheck

**Comment by Cyber-SiKu** [comment]
cicheck

**Comment by Cyber-SiKu** [comment]
cicheck

**Comment by Cyber-SiKu** [comment]
cicheck

**Pull request merged**